service/src/main/java/bio/terra/workspace/service/resource/controlled/cloud/azure/KubernetesClientProvider.java

```diff
@@ -145,13 +147,11 @@ public StepResult stepResultFromException(ApiException exception, HttpStatus...
         .orElseGet(StepResult::getStepResultSuccess);
   }
 
-  public ApiAzureLandingZoneDeployedResource getClusterResource(UUID workspaceId) {
+  public Optional<ApiAzureLandingZoneDeployedResource> getClusterResource(UUID workspaceId) {
     var bearerToken = new BearerToken(samService.getWsmServiceAccountToken());
     UUID landingZoneId =
         landingZoneApiDispatch.getLandingZoneId(
             bearerToken, workspaceService.getWorkspace(workspaceId));
-    return landingZoneApiDispatch
-        .getSharedKubernetesCluster(bearerToken, landingZoneId)
-        .orElseThrow(() -> new RuntimeException("No shared cluster found"));
+    return landingZoneApiDispatch.getSharedKubernetesCluster(bearerToken, landingZoneId);
```

Comment:
This is changed to return the `Optional`, so that downstream consumers can handle the missing resource differently, instead of trying to catch a runtime exception or come up with a new exception type.